### PR TITLE
[camera] update api reference docs to include not supported for globe

### PR DESCRIPTION
**Pull request by tobrun**
Update API reference docs to include not supported for globe projection.
The engine will return a no-op result.

## Pull request checklist:
 - [ ] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-api-*` CI steps might fail.
 - [x] Update [CHANGELOG.md](../CHANGELOG.md) or use the label 'skip changelog', otherwise `check changelog` CI step will fail.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

### CODE REVIEW

File: sdk-base/src/main/java/com/mapbox/maps/plugin/delegates/MapCameraManagerDelegate.kt [hunk 1]
```diff
@@ -19,6 +19,8 @@ interface MapCameraManagerDelegate {
   /**
    * Convenience method that returns the camera options object for given arguments
    *
+   * This API isn't supported by Globe projection.
```
Comment (yunikkk) [inline sdk-base/src/main/java/com/mapbox/maps/plugin/delegates/MapCameraManagerDelegate.kt:22]: What's an alternative though?

Comment (tobrun) [reply sdk-base/src/main/java/com/mapbox/maps/plugin/delegates/MapCameraManagerDelegate.kt:22]: there isn't :)

Comment (tobrun) [reply sdk-base/src/main/java/com/mapbox/maps/plugin/delegates/MapCameraManagerDelegate.kt:22]: we are looking to re-evaluate such APIs and potentially deprecating them

Comment (yunikkk) [reply sdk-base/src/main/java/com/mapbox/maps/plugin/delegates/MapCameraManagerDelegate.kt:22]: Maybe explicitly suggest to use Mercator then? Also - link to the Globe in javadoc, with `[Globe]` / `[Mercator]`

**Review by yunikkk — APPROVED** [review]

**Review by alexander-kulikovskii — APPROVED** [review]

File: sdk/src/main/java/com/mapbox/maps/MapboxMap.kt [hunk 2]
```diff
@@ -670,10 +686,13 @@ class MapboxMap :
    * Note that if the given `camera` shows the antimeridian, the returned wrapped [CoordinateBounds]
    * might not represent the minimum bounding box.
    *
+   * This API isn't supported by Globe projection and will return a no-op result.
```
Comment (pengdev) [inline sdk/src/main/java/com/mapbox/maps/MapboxMap.kt:689]: Shall we clarify no-op result? in this case this function will return a CoordinateBounds, is it nullable result or it's a CoordinateBounds with meaningless value?

**Review by pengdev — APPROVED** [review]
LGTM with 1 question.

**Review by kiryldz — APPROVED** [review]

**Pull request merged**